### ISSUE TITLE
Support form pagination

### ISSUE THREAD

**Issue by ei8fdb** [nextcloud/forms#357]
## Feature request
Support form pagination

## Rational
When designing large questionnaires/forms, using pagination (where the questionnaire is broken up into mulitple pages) helps the user by:
- breaking the tasks into sub-tasks allowing them to focus on one-a small number of questions at a time
- reducing the users cognitive load (less to intrepret)
- reducing distractions (scrolling up and down the page)

## UI

There's lots of ways of doing this. I can do some research into different approaches, but essentially it will:
- allow the user to create pages
- give it a title
- possible a short description

It would also be really great if it supported conditional logic (that's another ticket!)

There are many options on how to display this, but can help with designs if it helps.

## Priority
High

As a user researcher who creates questionnaires having questions that can gather ranking data is very important - however I can understand this is *complicated/a lot of work to implement*.

There's also a lot of ways to implement this so I think it requires time to discuss.

Thoughts?

**Comment by skjnldsv** [comment]
@jancborchardt

**Comment by jancborchardt** [comment]
Yes, we should absolutely support it. :)

@ei8fdb can you show screenshots of how Google Forms does it?

**Comment by ei8fdb** [comment]
> @ei8fdb can you show screenshots of how Google Forms does it?

Sure, I'll look at GoogForms and others. Goog's approach is OK, but it often trips me up.

**Comment by agentlibre** [comment]
It is probably next level, but in the same vein, I love this very smooth design from TypeForm if you haven't seen it yet: 

- https://klimaschutzschweiz.typeform.com/to/VKXjPk

 (specific example, I haven't found the generic feature yet, I'll make to edit my post if I do)

**Comment by jancborchardt** [comment]
> It is probably next level, but in the same vein, I love this very smooth design from TypeForm if you haven't seen it yet:

Yeah, we analyzed that together with Google Forms – and to be honest the Typeform design, while looking nice, has several issues, like you have no idea how long a form actually is and it can feel a lot.
So Google Forms’ simple default "1 page" is quite reassuring and easy, which is what we want to go for as well.

**Comment by jospoortvliet** [comment]
> > 
> 
> Yeah, we analyzed that together with Google Forms – and to be honest the Typeform design, while looking nice, has several issues, like you have no idea how long a form actually is and it can feel a lot.
> So Google Forms’ simple default "1 page" is quite reassuring and easy, which is what we want to go for as well.

I do bump into statistics that [multi-page forms are 300% more likely to be filled in by people.](https://wpforms.com/online-form-statistics-facts/) I know, lies, damned lies and statistics, but I personally also are much more likely to go through steps if it is multi-page than one long list. Esp if it has a progress bar and a time indicator, it just files more friendly. With big surveys especially, it also allows grouping and it really feels shorter that way. Just my cents

**Comment by jancborchardt** [comment]
Yep, @jospoortvliet of course we should have the ability to group questions and break stuff into pages. :) I was specifically referring to Typeform’s design of "1 page per question" which is very specific and doesn’t give a good overview.

**Comment by RenataGegaj** [comment]
From the user testings: There was only one request for this feature(that participant had a background in research). A big +1 from my side as well for this. I often need it for my survey work. 
I quite like the SurveyMonkey approach on this. Basically, the user is presented with questions from a section in one screen but they become active on scroll. That way they don't compete for attention or cause confusion. There is also a progress bar at the bottom for visual presentation, supported by numbers, to reassure the user(example below). The only thing missing here is the section title.
 
<img width="1435" alt="Screenshot 2022-01-21 at 15 10 07" src="https://user-images.githubusercontent.com/7413721/150543199-6e25eb55-57db-4099-a624-76f518ebe336.png">

**Comment by JOduMonT** [comment]
I think I have a similar and compatible idea, I was thinking if it could be possible, via a cookie, or you know more than me, to save the status of an ongoing filling form. Which, in that case, could be when the person pass to the next question (I think LimeSurvey do that)

My point is I have an intake form where I ask people to think about it and take their time (2-3days); of course they could leave the tab open and the computer on, but it would be awesome to generate a cookie with a URL, so they could refer to it when they want to continue where they left.

**Comment by foxhoundv** [comment]
This would greatly help my use for it. For work, when a tech is on site there are a bit of things possible to fill out. If they are able to show certain questions from different pages dependent on the previous answer... It could potentially make a form with 50 questions reduced to maybe 15 depending on what options were selected and then what pages weren't shown because they weren't needed.
